src/ddmd/mtype.d

```diff
@@ -2717,6 +2717,23 @@ extern (C++) abstract class Type : RootObject
      */
     final Expression noMember(Scope* sc, Expression e, Identifier ident, int flag)
     {
+        //printf("Type.noMember(e: %s ident: %s flag: %d)\n", e.toChars(), ident.toChars(), flag);
+
+        static __gshared int nest;
+
+        static Expression returnExp(Expression e)
```

Comment:
I suggest comparing the EH output with gdc/ldc. (Chandler Carruth told me that EH essentially disables the llvm optimize.)

The problem is in the front end. If the code is nothrow, EH should not be used at all. But the front end hasn't evolved that far, and even if it had, none of the dmd front end code is marked nothrow.